geolocator/android/src/main/java/com/baseflow/geolocator/permission/PermissionManager.java

```diff
@@ -0,0 +1,119 @@
+package com.baseflow.geolocator.permission;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+
+import com.baseflow.geolocator.errors.ErrorCallback;
+import com.baseflow.geolocator.errors.PermissionUndefinedException;
+
+import java.util.Map;
+
+import io.flutter.plugin.common.PluginRegistry;
+
+public class PermissionManager {
+    @FunctionalInterface
+    public interface ActivityRegistry {
+        void addListener(PluginRegistry.ActivityResultListener handler);
+    }
+
+    @FunctionalInterface
+    public interface PermissionRegistry {
+        void addListener(PluginRegistry.RequestPermissionsResultListener handler);
+    }
+
+    @FunctionalInterface
+    public interface RequestPermissionsSuccessCallback {
+        void onSuccess(Map<Integer, Integer> results);
+    }
+
+    @FunctionalInterface
+    public interface CheckPermissionsSuccessCallback {
+        void onSuccess(int permissionStatus);
+    }
```

Comment:
I think you could make one interface here since we will not return multiple permission states. We will always just return the permission status for the location permission. So maybe do something like:

```suggestion
    @FunctionalInterface
    public interface PermissionsSuccessCallback {
        void onSuccess(int permissionStatus);
    }
```

Reply:
I guess we should return the enum than as well instead of an int?
```suggestion
    @FunctionalInterface
    public interface PermissionsSuccessCallback {
        void onSuccess(LocationPermission permissionStatus);
    }
```

Reply:
yes I think that would be good. That would mean that we need to do the conversion to int in the `MethodHandlerImpl` but I think that is correct since that is the class that is responsible for communication over the method channel.